gcp/gcpcloud/gcpcloud.go

```diff
@@ -37,3 +37,17 @@ var Services = wire.NewSet(
 	gcp.NewHTTPClient,
 	runtimeconfigurator.Set,
 	sdserver.Set)
+
+// GAE is a Wire provider set, similar to the GCP Wire provider
+// set except that GAEServices is used instead of Services.
+var GAE = wire.NewSet(GAEServices, gcp.DefaultIdentity)
+
+// GAEServices is a Wire provider set, similar to Services
+// except that cloudmysql.OpenGAE is used instead of cloudmysql.Open.
+var GAEServices = wire.NewSet(
```

Comment:
I think you could also factor out the common ones, e.g.:

var sharedServices = wire.NewSet(cloudmysql.CertSourceSet, ...)
var Services = wire.NewSet(sharedServices, cloudmysql.Open)
var GAEServices = wire.NewSet(sharedServices, cloudmysql.OpenGAE)

Not sure if this is better though.

Reply:
Done.